src/sst/core/baseComponent.h

```diff
@@ -272,6 +279,51 @@ class BaseComponent {
     }
 
 
+
+    template <typename T>
+    Statistics::Statistic<T>* createStatistic(SST::Params& params, StatisticId_t id,
+                                              const std::string& name, const std::string& statSubId){
+
+      /* I would prefer to avoid this std::function with dynamic cast,
+       * but the code is just a lot cleaner and avoids many unnecessary template instantiations
+       * doing it this way. At some point in the future, we would need to clean up
+       * the rule around enabling all statistics to make this better
+       */
+
+      StatCreateFunction create = [=](BaseComponent* comp, Statistics::StatisticProcessingEngine* engine,
+                                      const std::string& type, const std::string& name, const std::string& subId,
+                                      SST::Params& params)
+          -> Statistics::StatisticBase* {
+        return engine->createStatistic<T>(comp, type, name, subId, params);
+      };
+
+      // We follow two distinct paths depending on if it is enable all, verus explicitly enabled
+      // Enable all is "scoped" to the (sub)component
+      // Explicitly enabled stats are assigned component-unique IDs and can be shared across subcomponents
+      // so creation and management happens in the parent component
+      Statistics::StatisticBase* base_stat = id == STATALL_ID
+         ? createEnabledAllStatistic(params, name, statSubId, std::move(create))
+         : getParentComponent()->createExplicitlyEnabledStatistic(params, id, name, statSubId, std::move(create));
+
+      // Ugh, dynamic casts hurt my eyes, but I must do this
+      auto* statistic = dynamic_cast<Statistics::Statistic<T>*>(base_stat);
+      if (statistic){
+         return statistic;
+      } else {
+        fatal(__LINE__, __FILE__, "createStatistic", 1,
+             "failed to cast created statistic '%s' to expected type",
+             name.c_str());
+        return nullptr; //avoid compiler warnings
+      }
+    }
+
+
+    template <typename T>
+    Statistics::Statistic<T>* createNullStatistic(SST::Params& params, const std::string& name, const std::string& statSubId = ""){
+      auto* engine = Statistics::StatisticProcessingEngine::getInstance();
+      return engine->createStatistic<T>(my_info->component, "sst.NullStatistic", name, statSubId, params);
```

Comment:
I know we’ve always done it this way, but do we need to create a new NullStatistic for each stat that isn’t enabled, or can we just have a single NullStatistic object that we give to everyone.  There wouldn’t be any data associated with it, just an addData() function that immediately returns.  Not sure this is high priority, because it probably wouldn’t save much memory.

Reply:
We could potentially modify NullStatistic to avoid some of the overheads in the `addData`function, but I think it would be too complicated for the benefit to change to a single `NullStatistic`